arduino-ide-extension/src/browser/widgets/sketchbook/sketchbook-tree-model.ts

```diff
@@ -15,14 +20,102 @@ export class SketchbookTreeModel extends FileTreeModel {
     @inject(ArduinoPreferences)
     protected readonly arduinoPreferences: ArduinoPreferences;
 
+    @inject(CommandRegistry)
+    protected commandRegistry: CommandRegistry;
+
     @inject(ConfigService)
     protected readonly configService: ConfigService;
 
+    @inject(OpenerService)
+    protected readonly openerService: OpenerService;
+
+    @inject(SketchesServiceClientImpl)
+    protected readonly sketchServiceClient: SketchesServiceClientImpl;
+
     async updateRoot(): Promise<void> {
         const config = await this.configService.getConfiguration();
         const fileStat = await this.fileService.resolve(new URI(config.sketchDirUri));
         const showAllFiles = this.arduinoPreferences['arduino.sketchbook.showAllFiles'];
         this.tree.root = SketchbookTree.RootNode.create(fileStat, showAllFiles);
     }
 
+    async openNode(raw?: TreeNode | undefined) {
+        const node = raw || this.selectedNodes[0];
+
+        if (!node) { return }
+
+        // if it's a sketch dir, or a file from another sketch, open in new window
+        const sketchRoot = await this.getSketchRoot(node);
+        if (sketchRoot) {
+            this.commandRegistry.executeCommand(SketchbookCommands.OPEN_NEW_WINDOW.id, { node: sketchRoot })
+            return;
+        }
+
+        if (node) {
+            this.doOpenNode(node);
+            this.onOpenNodeEmitter.fire(node);
+        }
+    }
+
+    // selectNode gets called when the user single-clicks on an item
```

Comment:
> selectNode gets called when the user single-clicks on an item

What does this `previewNode` suppose to do? Can you please explain?

Reply:
it should open a file straight away. I can probably use a better name, as preview is a different functionality which I tested but was not what requested by Gianni/Ubi

Reply:
> it should open a file straight away

What should I do to hit the breakpoint after the `return`.?

Reply:
I refactored the code and made it more clear